metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart

```diff
@@ -0,0 +1,930 @@
+        );
+        responses.addRunsPages([runsPage]);
+
+        responses.addRunArtifactsPages([defaultRunArtifactPage]);
+
+        final firstBuild = createBuildData(buildNumber: 1);
+        final expected = createBuildDatas(buildNumbers: [4, 3, 2]);
+
+        whenFetchRuns().thenAnswer((_) => responses.fetchWorkflowRuns());
+
+        final result = adapter.fetchBuildsAfter(
+          defaultWorkflowIdentifier,
+          firstBuild,
+        );
+
+        expect(result, completion(equals(expected)));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() returns an empty list if there are no new builds",
+      () {
+        final runsPage = WorkflowRunsPage(
+          values: createWorkflowRuns(runNumbers: [4, 3, 2, 1]),
+        );
+        responses.addRunsPages([runsPage]);
+
+        responses.addRunArtifactsPages([defaultRunArtifactPage]);
+
+        final firstBuild = createBuildData(buildNumber: 5);
+        final expected = [];
+
+        whenFetchRuns().thenAnswer((_) => responses.fetchWorkflowRuns());
+
+        final result = adapter.fetchBuildsAfter(
+          defaultWorkflowIdentifier,
+          firstBuild,
+        );
+
+        expect(result, completion(equals(expected)));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() returns new builds if the given build was deleted",
+      () {
+        final runsPage = WorkflowRunsPage(
+          values: createWorkflowRuns(runNumbers: [7, 6, 5, 4, 2, 1]),
+        );
+        responses.addRunsPages([runsPage]);
+
+        responses.addRunArtifactsPages([defaultRunArtifactPage]);
+
+        final firstBuild = createBuildData(buildNumber: 3);
+        final expected = createBuildDatas(buildNumbers: [7, 6, 5, 4]);
+
+        whenFetchRuns().thenAnswer((_) => responses.fetchWorkflowRuns());
+
+        final result = adapter.fetchBuildsAfter(
+          defaultWorkflowIdentifier,
+          firstBuild,
+        );
+
+        expect(result, completion(equals(expected)));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches coverage for each build",
+      () async {
+        final runsPage = WorkflowRunsPage(
+          values: createWorkflowRuns(runNumbers: [4, 3, 2, 1]),
+        );
+        responses.addRunsPages([runsPage]);
+
+        responses.addRunArtifactsPages([defaultRunArtifactPage]);
+
+        final firstBuild = createBuildData(buildNumber: 1);
+
+        final expected = [defaultCoverage, defaultCoverage, defaultCoverage];
+
+        whenFetchRuns().thenAnswer((_) => responses.fetchWorkflowRuns());
+
+        final result = await adapter.fetchBuildsAfter(
+          defaultWorkflowIdentifier,
+          firstBuild,
+        );
+
+        final actualCoverage =
+            result.map((buildData) => buildData.coverage).toList();
+
+        expect(actualCoverage, equals(expected));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches coverage for each build if the coverage artifact is in next run artifacts pages",
```

Comment:
The same comment here and below as for appropriate test cases for `.fetchBuilds()`.